proxylib/memcached/text/parser.go

```diff
@@ -0,0 +1,468 @@
+		} else if p.isCommandStorage(meta.command) {
+			// storage commands
+			meta.keys = tokens[1:2]
+			nBytes, err := strconv.Atoi(string(tokens[4]))
+			if err != nil {
+				log.Error("Failed to parse storage payload length")
+				return proxylib.ERROR, 0
+			}
+			// 2 additional bytes for terminating linefeed
+			frameLength = linefeed + 2 + nBytes + 2
+
+			if meta.command[0] == 'c' { //storage command is "cas"
+				noreply = len(tokens) == 7
+			} else {
+				noreply = len(tokens) == 6
+			}
+		} else if p.isCommandDelete(meta.command) {
+			meta.keys = tokens[1:2]
+			noreply = len(tokens) == 3
+			frameLength = linefeed + 2
+		} else if p.isCommandIncrDecr(meta.command) {
+			meta.keys = tokens[1:2]
+			noreply = len(tokens) == 4
+			frameLength = linefeed + 2
+		} else if bytes.Equal(meta.command, []byte("touch")) {
+			meta.keys = tokens[1:2]
+			noreply = len(tokens) == 4
+			frameLength = linefeed + 2
+		} else if bytes.Equal(meta.command, []byte("slabs")) ||
+			bytes.Equal(meta.command, []byte("lru")) ||
+			bytes.Equal(meta.command, []byte("lru_crawler")) ||
+			bytes.Equal(meta.command, []byte("stats")) ||
+			bytes.Equal(meta.command, []byte("version")) ||
+			bytes.Equal(meta.command, []byte("misbehave")) {
+
+			meta.keys = [][]byte{}
+			frameLength = linefeed + 2
+		} else if bytes.Equal(meta.command, []byte("flush_all")) ||
+			bytes.Equal(meta.command, []byte("cache_memlimit")) {
+			meta.keys = [][]byte{}
+			noreply = bytes.Equal(tokens[len(tokens)-1], []byte("noreply"))
+			frameLength = linefeed + 2
+		} else if bytes.Equal(meta.command, []byte("quit")) {
+			meta.keys = [][]byte{}
+			noreply = true
+			frameLength = linefeed + 2
+		} else {
+			log.Error("Could not parse text memcache frame")
+			return proxylib.ERROR, 0
+		}
+		logEntry := &cilium.LogEntry_GenericL7{
+			&cilium.L7LogEntry{
+				Proto: "textmemcached",
+				Fields: map[string]string{
+					"command": string(meta.command),
+					"keys":    string(bytes.Join(meta.keys, []byte(", "))),
+				},
+			},
+		}
+
+		p.requestCount++
+		r := &replyIntent{
+			requestID: p.requestCount,
+			command:   meta.command,
+		}
+
+		matches := p.connection.Matches(meta)
+
+		if matches {
+			r.denied = false
+			p.replyQueue = append(p.replyQueue, r)
+			p.connection.Log(cilium.EntryType_Request, logEntry)
+			return proxylib.PASS, frameLength
+		}
+
+		r.denied = true
+		if !noreply {
+			p.injectDeniedMessage()
+			if p.requestCount == p.replyCount+1 {
+			} else {
+				p.replyQueue = append(p.replyQueue, r)
+			}
+		}
+		p.connection.Log(cilium.EntryType_Denied, logEntry)
+		return proxylib.DROP, frameLength
+	} else {
+		//reply
+		log.Debugf("reply, parsing to figure out if we have it all")
+
+		intent := p.replyQueue[0]
+
+		logEntry := &cilium.LogEntry_GenericL7{
+			&cilium.L7LogEntry{
+				Proto: "textmemcached",
+				Fields: map[string]string{
+					"command": string(intent.command),
+				},
+			},
+		}
+
+		if p.isErrorReply(tokens[0]) ||
```

Comment:
`switch{ case/ case/`

Reply:
Again, which value should I switch on here?

Reply:
none, see https://tour.golang.org/flowcontrol/11